### PR TITLE
fix: typescript React import

### LINKED ISSUE #3178: React should import * as
image

shouldn't use `import React from 'react'`

although I can use `"allowSyntheticDefaultImports": true` in `tsconfig.json` to avoid the problem

### PR DESCRIPTION
**Pull request by ycjcl868**
Close #3178.

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/3179)

### CODE REVIEW

**Comment by doxiaodong** [comment]
Not enough since `rmc-xxx` isn't be changed

**Comment by ycjcl868** [comment]
@doxiaodong There have no effect for developers using `antd-mobile`.

**Comment by doxiaodong** [comment]
@ycjcl868 some declare types is defined by rmc-xxx

**Comment by ycjcl868** [comment]
What files ?

**Comment by doxiaodong** [comment]
Thank you . also change `import * as PropTypes from 'prop-types';` ?

**Comment by ycjcl868** [comment]
@doxiaodong It doesn't matter to developers using `antd-mobile`, it's just for `antd-mobile` developers. In addition, `allowSyntheticDefaultImports` was enabled in `antd-mobile` tsconfig.json, so this is no need to change `PropTypes`.

**Comment by doxiaodong** [comment]
Also change the `prop-types`, thanks

**Comment by doxiaodong** [comment]
`prop-types` will be used by user since typescript will identify `static propTypes` in component, so it's need to change it.

**Comment by ycjcl868** [comment]
ok,

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "babel-runtime": "6.x",
     "classnames": "^2.2.1",
     "normalize.css": "^7.0.0",
+    "prop-types": "^15.7.2",
```
Comment (doxiaodong) [inline package.json]: remove this. the `prop-types` should be chosen by users

**Comment by doxiaodong** [comment]
https://github.com/ant-design/ant-design-mobile/pull/3186  这个先合吧， antd 修正后导致  compile 不过了

**Pull request merged**